spec/validations_spec.rb

```diff
@@ -1,19 +1,151 @@
+require 'spec_helper'
 require_relative '../lib/validations'
+require_relative '../lib/ship'
 
 RSpec.describe Validations do
 
   let(:validations) {Validations.new}
+  let(:destroyer) {Ship.new("destroyer", 2)}
 
-  it "raises error for wrong input for ship number" do
-    output = validations.validate_ship_number("5")
+  describe "checks player's entered number corresponds to a ship" do
+    it "returns :invalid_ship_number for invalid number" do
+      output = validations.validate_ship_number("5")
 
-    expect(output).to eq(:error)
+      expect(output).to eq(:invalid_ship_number)
+    end
+
+    it "returns :invalid_ship_number for invalid input type" do
+      output = validations.validate_ship_number("hello")
+
+      expect(output).to eq(:invalid_ship_number)
+    end
+
+    it "returns :valid_ship_number for valid input" do
+      output = validations.validate_ship_number("2")
+
+      expect(output).to eq(:valid_ship_number)
+    end
+  end
+
+  describe "checks player's inputs for ship to place are all valid" do
+    it "returns :invalid_ship_position if 1st coordinate is not number" do
+      ui_output = {:x => "a", :y => "b", :orientation => :horizontal}
+      output = validations.validate_position_for_ship(destroyer, ui_output)
+
+      expect(output).to eq(:invalid_ship_position)
+    end
+
+    it "returns :invalid_ship_position if 1st coordinate is not valid number" do
+      ui_output = {:x => "100", :y => "b", :orientation => :horizontal}
+      output = validations.validate_position_for_ship(destroyer, ui_output)
+
+      expect(output).to eq(:invalid_ship_position)
+    end
+
+    it "returns :invalid_ship_position if 2nd coordinate is a number" do
+      ui_output = {:x => "1", :y => "2", :orientation => :horizontal}
+      output = validations.validate_position_for_ship(destroyer, ui_output)
+
+      expect(output).to eq(:invalid_ship_position)
+    end
+
+    it "returns :invalid_ship_position if 2nd coordinate is an invalid letter" do
+      ui_output = {:x => "1", :y => "x", :orientation => :horizontal}
+      output = validations.validate_position_for_ship(destroyer, ui_output)
+
+      expect(output).to eq(:invalid_ship_position)
+    end
+
+    it "returns :invalid_ship_position if 3rd coordinate (orientation) is not valid" do
+      ui_output = {:x => "1", :y => "b", :orientation => "c"}
+      output = validations.validate_position_for_ship(destroyer, ui_output)
+
+      expect(output).to eq(:invalid_ship_position)
+    end
+
+    it "returns :invalid_ship_position if all inputs are invalid" do
```

Comment:
Not sure if you need this test after you've already checked individually for each invalid input. I think you can probably remove it. It's good to have everything covered by tests but it's also good to have as few tests as possible. Because the more code (including tests) you have, the more code you have to maintain and update when things change. So it's a bit of a balance from having enough tests but not too many, especially if they are somewhat duplicated.